### PR TITLE
Mattlib/rolling 0.1.0 stability

### PR DESCRIPTION
**Pull request by superyuuki**

### CODE REVIEW

**Comment by superyuuki** [comment]
@cpostbitbuckets merge priority asap

File: src/main/java/org/bitbuckets/drive/controlsds/falcon/Falcon500DriveControlSDSSetup.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
  * <p>
  * really fucking simple because a drivecontrol is super simple LMAO
```
Comment (cpostbitbuckets) [inline src/main/java/org/bitbuckets/drive/controlsds/falcon/Falcon500DriveControlSDSSetup.java:31]: clean this up please

File: src/main/java/org/bitbuckets/arm/IArmControl.java [hunk 2]
```diff
@@ -4,10 +4,8 @@
...
 public interface IArmControl {
 
-    Mechanism2d robotArmRootNode();
-
-    Mechanism2d robotArmLigamentNode();
-
-    Mechanism2d robotArmJointNode();
+    //Mechanism2d robotArmRootNode();
+    //Mechanism2d robotArmLigamentNode();
+    //Mechanism2d robotArmJointNode();
```
Comment (cpostbitbuckets) [inline src/main/java/org/bitbuckets/arm/IArmControl.java:9]: just delete them?

File: src/main/java/org/bitbuckets/lib/core/LoopDriver.java [hunk 3]
```diff
@@ -40,7 +40,7 @@ public void registerLoopPeriodic(int parentId, Runnable loop, int period_ms) {
 
     }
 
-    public void runAlways() {
+    public void runPeriodic() {
```
Comment (cpostbitbuckets) [inline src/main/java/org/bitbuckets/lib/core/LoopDriver.java:43]: do we have a runAlways somewhere else for logging when disabled? like mechanism sensors?

File: src/main/java/org/bitbuckets/lib/hardware/IEncoder.java [hunk 4]
```diff
@@ -3,28 +3,58 @@
...
  */
 public interface IEncoder extends IRaw {
 
     //actual methods
 
+    /**
+     *
+     * @return a coefficient representing the conversion of rotations of the motor
+     * to rotations of whatever mechanism the motor is meant to be driving
+     */
```
Comment (cpostbitbuckets) [inline src/main/java/org/bitbuckets/lib/hardware/IEncoder.java:17]: nice. docs!

**Review by cpostbitbuckets — CHANGES_REQUESTED** [review]
Lots here, I did my best to review it. Please remove the swearing (I know it's not from this commit, but still)

File: src/main/java/org/bitbuckets/drive/controlsds/falcon/Falcon500DriveControlSDSSetup.java (hunk 1 above)
Comment (superyuuki) [reply src/main/java/org/bitbuckets/drive/controlsds/falcon/Falcon500DriveControlSDSSetup.java:31]: Yes

File: src/main/java/org/bitbuckets/lib/core/LoopDriver.java (hunk 3 above)
Comment (superyuuki) [reply src/main/java/org/bitbuckets/lib/core/LoopDriver.java:43]: Should clarify that this is always, just wanted to distinguish it from sim

File: src/main/java/org/bitbuckets/lib/hardware/IEncoder.java (hunk 4 above)
Comment (superyuuki) [reply src/main/java/org/bitbuckets/lib/hardware/IEncoder.java:17]: Yes, with more to come

File: src/main/java/org/bitbuckets/lib/core/LoopDriver.java (hunk 3 above)
Comment (cpostbitbuckets) [reply src/main/java/org/bitbuckets/lib/core/LoopDriver.java:43]: 👍

**Review by cpostbitbuckets — APPROVED** [review]

**Review by cpostbitbuckets — CHANGES_REQUESTED** [review]
Crashes in the sim:

```
Error at org.bitbuckets.lib.util.MockingUtil.buddy(MockingUtil.java:48): [BUCKET] Critical exception during setup: java.lang.InstantiationException: org.bitbuckets.drive.controlsds.DriveControlSDS$ByteBuddy$Rq4z3bsX
	at org.bitbuckets.lib.util.MockingUtil.buddy(MockingUtil.java:48)
	at org.bitbuckets.robot.RobotSetup.build(RobotSetup.java:125)
	at org.bitbuckets.bootstrap.Robot.robotInit(Robot.java:64)
	at org.littletonrobotics.junction.LoggedRobot.startCompetition(LoggedRobot.java:59)
	at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:343)
	at edu.wpi.first.wpilibj.RobotBase.lambda$startRobot$0(RobotBase.java:413)
	at java.base/java.lang.Thread.run(Thread.java:833)

Error at java.base/java.lang.Class.newInstance(Class.java:639): Unhandled exception: java.lang.InstantiationException: org.bitbuckets.drive.controlsds.DriveControlSDS$ByteBuddy$Rq4z3bsX
	at java.base/java.lang.Class.newInstance(Class.java:639)
	at org.bitbuckets.lib.util.MockingUtil.buddy(MockingUtil.java:45)
	at org.bitbuckets.robot.RobotSetup.build(RobotSetup.java:125)
	at org.bitbuckets.bootstrap.Robot.robotInit(Robot.java:64)
	at org.littletonrobotics.junction.LoggedRobot.startCompetition(LoggedRobot.java:59)
	at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:343)
	at edu.wpi.first.wpilibj.RobotBase.lambda$startRobot$0(RobotBase.java:413)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Review by cpostbitbuckets — APPROVED** [review]

**Pull request merged**